src/main/java/com/gft/component/SimpleMovingAverage.java

```diff
@@ -0,0 +1,67 @@
+package com.gft.component;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Date;
+
+import org.springframework.stereotype.Component;
+
+import com.gft.model.Action;
+import com.gft.model.db.Stock;
+import com.gft.model.db.StockHistory;
+import com.gft.repository.HistoryDAO;
+
+@Component
+public class SimpleMovingAverage implements PredictionAlgorithm {
```

Comment:
Klasa sama w sobie nie powinna być komponentem, gdyż domyślnie jest to singleton - czyli tylko 1 egzemplarz. Chcielibyśmy, aby możliwe było wstrzyknięcie przez springa 3-ech sparametryzowanych algorytmów. Można tego dokonać tworząć w klasie Application następujące metody:

@Bean(name = "simpleMovingAverage5")
public PredictionAlgorithm simpleMovingAverage5() {
     return new SimpleMovingAverage(5);
} i to samo dla 10, 5 etc.

Można potem wstrzyknąć konretnego beana w nastepujący sposób po id/name:

@Autowired @Qualifier("simpleMovingAverage5")
PredictionAlgorithm simpleMovingAverage5;

albo 

@Resource(name="simpleMovingAverage5")
PredictionAlgorithm simpleMovingAverage5;